domain/secretbackend/doc.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package secretbackend provides domain types and services for managing secret
+// backend definitions and model-level backend selection.
+//
+// The secret backend domain is responsible for:
+//   - Defining and validating secret backend records (identity, type, config).
+//   - Creating, updating, listing, and deleting backends, with safeguards for
+//     immutability and in-use protection in the state layer.
+//   - Tracking rotation metadata (token rotate interval and next rotation time)
+//     and emitting changes consumable by watchers used by rotation workers.
+//   - Managing per-model backend selection, the default model backend is
+//     automatically resolved to the most appropriate built-in backend based on
+//     model type (IAAS or CAAS).
+//   - Restoring the default backend selection when a user-defined backend is
+//     destroyed, respectfully the model type.
+//
+// Key concepts
+//
+//   - Backend identity: a backend can be addressed by UUID or by human-readable
+//     name. Only one is required for lookups/updates.
+//   - Backend type: backend types are either controller for the built-in backend
+//     into juju controller, kubernetes for the Kubernetes provider, or vault for
```

Comment:
`... for the backend built into the Juju contoller...`

The backend types should be enclosed in back ticks